aptos-move/move-examples/fungible_asset/sources/coin_example.move

```diff
@@ -0,0 +1,136 @@
+module aptos_framework::coin_example {
+    use aptos_framework::object;
+    use std::string::String;
+    use aptos_framework::object::{generate_extend_ref, Object, disable_ungated_transfer, generate_transfer_ref};
+    use fungible_asset::managed_fungible_asset::{initialize_managing_capabilities, mint_by_asset_owner, freeze_by_asset_owner, burn_by_asset_owner};
+    use std::signer::address_of;
+    use fungible_asset::fungible_asset;
+    use std::error;
+    use aptos_std::smart_table;
+    use aptos_std::smart_table::SmartTable;
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+    struct Coin has key {
+        name: String,
+        /// Symbol of the coin, usually a shorter version of the name.
+        /// For example, Singapore Dollar is SGD.
+        symbol: String,
+        /// Number of decimals used to get its user representation.
+        /// For example, if `decimals` equals `2`, a balance of `505` coins should
+        /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+        decimals: u8
+    }
+
+    struct CoinIndexer has key {
+        index: smart_table::SmartTable<String, Object<Coin>>
+    }
+
+    public entry fun create_coin(creator: &signer, name: String, symbol: String, decimals: u8) acquires CoinIndexer {
+        let creator_ref = object::create_object_from_account(creator);
+        let object_signer = object::generate_signer(&creator_ref);
+        disable_ungated_transfer(&generate_transfer_ref(&creator_ref));
+        move_to(&object_signer, Coin {
+            name,
+            symbol,
+            decimals
+        });
+
+        initialize_managing_capabilities(&generate_extend_ref(&creator_ref), 0 /* no max supply */);
+        let coin_obj = object::object_from_constructor_ref<Coin>(&creator_ref);
+        if (!exists<CoinIndexer>(address_of(creator))) {
```

Comment:
Can you add a TODO here to revisit this regarding create_named_object vs create_object_from_account?